### PR TITLE
Mockery options

### PR DESCRIPTION
**Pull request by Yashk767**
# Description

Used Mockery for testing options.go and client_methods.go
- Replaced the tests for options.go with tests written using mockery
- Added tests for client_methods.go

Fixes: #397 #396

### CODE REVIEW

File: utils/math_test.go [hunk 1]
```diff
@@ -122,65 +122,65 @@ func TestConvertToNumber(t *testing.T) {
...
-	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			if got := MultiplyFloatAndBigInt(tt.args.bigIntVal, tt.args.floatingVal); !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("MultiplyFloatAndBigInt() = %v, want %v", got, tt.want)
-			}
-		})
-	}
-}
+//func TestMultiplyFloatAndBigInt(t *testing.T) {
```
Comment (ashish10677) [inline utils/math_test.go:125]: Why has this been commented out?

File: utils/api_test.go [hunk 2]
```diff
@@ -190,15 +190,15 @@ func TestGetDataFromHTML(t *testing.T) {
...
-		{
-			name: "Test 2: Test data from coin gecko",
-			args: args{
-				url:      "https://www.coingecko.com/en",
-				selector: `table tbody tr td a[href="/en/coins/bitcoin"]`,
-			},
-			want:    "\nBTC\n",
-			wantErr: false,
-		},
+		//{
```
Comment (ashish10677) [inline utils/api_test.go:193]: Remove this entirely, no need to comment it out.

**Review by ashish10677 — CHANGES_REQUESTED** [review]

File: utils/math_test.go (hunk 1 above)
Comment (Yashk767) [reply utils/math_test.go:125]: Oh we changed the method caller there so I commented that time. I will uncomment it now.

**Comment by ashish10677** [comment]
@Yashk767 please resolve conflicts.

**Review by ashish10677 — APPROVED** [review]

**Pull request merged**